### ISSUE TITLE
Some photos deformed on 'my district' page

### ISSUE THREAD

**Issue by adamboche** [JeffreyBLewis/WebVoteView#80]
http://128.97.229.160/district
Search 90024, look at Ted Lieu. Is his photo a little squashed?

**Comment by aaronrudkin** [comment]
He's squashed because his photo is unusually long and I coerce it to be roughly square. I can automagically detect who has unusually long photos (I believe I did the same thing for unusually wide photos a month or two ago) but automagically fixing them is going to be unreliable. I can do a center crop, but in Lieu's case we actually want a bottom crop.

**Comment by aaronrudkin** [comment]
I wrote some code that will identify every photo that is either:
1. Landscape (i.e. width>height)
2. Excessively long (i.e. height>1.4*width)

And move them to a quarantine, where I can do crop to 5:4 (i.e. proper portrait) aspect ratio. Of our photos, this affects approximately 2,250. The question is how to parameterize the crop. The naive way would be a center crop. The question is whether or not center cropping is the appropriate. 

Consider Kamala Harris, whose current photo is a seated portrait shot. Because her head is in the top center, a center crop would decapitate her. In the Ted Lieu case the framing would still be weird, but we wouldn't lose much of his face. The landscape cases are even weirder, because we could end up with people totally not in the picture.

I am trying to work out a content-aware crop mechanism.

**Comment by aaronrudkin** [comment]
I have a content-aware crop mechanism using [jwagner/smartcrop-cli](/jwagner/smartcrop-cli). It was a bit tricky because smartcrop-cli wants fixed pixel dimensions, not an aspect ratio, so I had to use bash to precompute the dimensions as I go along. Anyway, it seems to be working. It's not super fast but it's fast enough, given that we only need to do this once in a while.

I will close the issue when I have tested the photos and am sure that they are fixed, and then commit the fixed photos. I will make a Wiki document on how to use the smart cropper, and then put the source in the voteview_metadata repository.
